### PR TITLE
Added tests for Doctor class

### PR DESCRIPTION
**Pull request by chenlevy24**
Added tests for 'doctor' class and did minor changed in 'updateConfig', 'doctor' and 'configSpec'

### CODE REVIEW

File: packages/cli/src/lib/cmds/updateConfig.ts [hunk 1]
```diff
@@ -41,14 +41,14 @@ async function updateJdkPath(path: string, log: Log): Promise<boolean> {
...
-  newConfig.saveConfig(DEFAULT_CONFIG_FILE_PATH);
+  await newConfig.saveConfig(DEFAULT_CONFIG_FILE_PATH);
   return true;
 }
 
 export async function updateConfig(args: ParsedArgs, log: Log = new ConsoleLog('updateConfig')):
         Promise<boolean> {
   if (args.jdkPath) {
-    await updateJdkPath(args.JdkPath, log);
+    await updateJdkPath(args.jdkPath, log);
```
Comment (PEConn) [inline packages/cli/src/lib/cmds/updateConfig.ts:51]: Woohoo, it looks like our tests are already finding bugs!

File: packages/cli/src/spec/configSpec.ts [hunk 2]
```diff
@@ -30,14 +30,14 @@ const LEGACY_CONFIG_FOLDER = join(homedir(), '.llama-pack');
...
-  spyOn(JdkHelper, 'validatePath').and.returnValue(Promise.resolve(Result.ok('/path/to/jdk')));
-  spyOn(AndroidSdkTools, 'validatePath').and.returnValue(Promise.resolve(Result.ok(
-      '/path/to/android-sdk')));
-});
-
 describe('config', () => {
   describe('#loadOrCreateConfig', () => {
+    beforeAll(() => {
+      spyOn(JdkHelper, 'validatePath').and.returnValue(Promise.resolve(Result.ok('/path/to/jdk')));
+      spyOn(AndroidSdkTools, 'validatePath').and.returnValue(Promise.resolve(Result.ok(
```
Comment (PEConn) [inline packages/cli/src/spec/configSpec.ts:37]: Do you mind line breaking this at `.returnValue` - I know you didn't write this originally but I think it reads a bit better:

```
spyOn(AndroidSdkTools, 'validatePath').and
    .returnValue(Promise.resolve(Result.ok('/path/to/android-sdk')));
```

File: packages/cli/src/spec/DoctorSpec.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import {updateConfig} from '../lib/cmds/updateConfig';
+import {MockLog} from '@bubblewrap/core';
+import minimist = require('minimist');
+import {doctor} from '../lib/cmds/doctor';
+import {MockPrompt} from './mock/MockPrompt';
+import {enUS as messages} from '../lib/strings';
+
+describe('doctor', () => {
+  describe('#jdkDoctor', () => {
+    it('checks if a currect error is sent in case that the path given isn\'n valid', async () => {
```
Comment (PEConn) [inline packages/cli/src/spec/DoctorSpec.ts:28]: I'm fairly sure that "currect" is a typo, but I can't figure out what it's meant to be. Is it "correct"? In which case, what is a "correct error"?

File: packages/cli/src/spec/DoctorSpec.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      // Creates a mock file systes.
```
Comment (PEConn) [inline packages/cli/src/spec/DoctorSpec.ts:29]: s/systes/system

File: packages/cli/src/spec/DoctorSpec.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+      const parsedMockArgs = minimist(['--jdkPath', 'new/path/to/jdk']);
+      await updateConfig(parsedMockArgs, mockLog);
+      expect(await doctor(mockLog)).toBeFalse();
+      // Check that the correct message was sent.
+      const logErrors: Array<string> = mockLog.getReceivedData();
+      expect(logErrors[logErrors.length - 1]).toEqual(messages.jdkPathIsNotCorrect);
+      mock.restore();
+    });
+
+    it('checks if a currect error is sent in case that the jdk isn\'t supported', async () => {
```
Comment (PEConn) [inline packages/cli/src/spec/DoctorSpec.ts:58]: same here - oh do you mean "checks that the correct error message is sent"?

**Review by PEConn — COMMENTED** [review]
This is looking good, just a few nits

File: packages/cli/src/spec/configSpec.ts (hunk 2 above)
Comment (andreban) [reply packages/cli/src/spec/configSpec.ts:37]: Related and worth reading: https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break

File: packages/cli/src/spec/DoctorSpec.ts (hunk 5 above)
Comment (chenlevy24) [reply packages/cli/src/spec/DoctorSpec.ts:58]: Yup, I will fix it

File: packages/cli/src/spec/DoctorSpec.ts [hunk 6]
```diff
@@ -25,8 +25,9 @@ import {enUS as messages} from '../lib/strings';
 
 describe('doctor', () => {
   describe('#jdkDoctor', () => {
-    it('checks if a currect error is sent in case that the path given isn\'n valid', async () => {
-      // Creates a mock file systes.
+    it('checks that the expected error message is sent in case that the path given isn\'n' +
```
Comment (PEConn) [inline packages/cli/src/spec/DoctorSpec.ts:28]: isn't, not isn'n

File: packages/cli/src/spec/DoctorSpec.ts [hunk 7]
```diff
@@ -88,8 +90,9 @@ describe('doctor', () => {
   });
 
   describe('#androidSdkDoctor', () => {
-    it('checks if a currect error is sent in case that the path given isn\'n valid', async () => {
-      // Creates a mock file systes.
+    it('checks that the expected error message is sent in case that the path given isn\'n' +
```
Comment (PEConn) [inline packages/cli/src/spec/DoctorSpec.ts:93]: Same here

**Review by PEConn — APPROVED** [review]

**Review by andreban — APPROVED** [review]
LGTM, Thanks

**Pull request merged**